README.md

```diff
@@ -1,42 +1,93 @@
+[![CI](https://img.shields.io/github/workflow/status/johnpryan/shelf_gzip/Dart%20CI/master?logo=github-actions&logoColor=white)](https://github.com/johnpryan/shelf_gzip/actions)
+[![GitHub Tag](https://img.shields.io/github/v/tag/johnpryan/shelf_gzip?logo=git&logoColor=white)](https://github.com/johnpryan/shelf_gzip/releases)
+[![New Commits](https://img.shields.io/github/commits-since/johnpryan/shelf_gzip/latest?logo=git&logoColor=white)](https://github.com/johnpryan/shelf_gzip/network)
+[![Last Commits](https://img.shields.io/github/last-commit/johnpryan/shelf_gzip?logo=git&logoColor=white)](https://github.com/johnpryan/shelf_gzip/commits/master)
+[![Pull Requests](https://img.shields.io/github/issues-pr/johnpryan/shelf_gzip?logo=github&logoColor=white)](https://github.com/johnpryan/shelf_gzip/pulls)
+[![Code size](https://img.shields.io/github/languages/code-size/johnpryan/shelf_gzip?logo=github&logoColor=white)](https://github.com/johnpryan/shelf_gzip)
+[![License](https://img.shields.io/github/license/johnpryan/shelf_gzip?logo=open-source-initiative&logoColor=green)](https://github.com/johnpryan/shelf_gzip/blob/master/LICENSE)
+
+Shelf middleware to GZIP encoding responses, with compression level and compression scope by content-type.
 
 ## Usage
 
 ```dart
+import 'dart:async' show runZonedGuarded;
 import 'dart:io';
-import 'dart:async' show runZoned;
+
 import 'package:path/path.dart' show join, dirname;
 import 'package:shelf/shelf.dart' as shelf;
 import 'package:shelf/shelf_io.dart' as shelf_io;
-import 'package:shelf_static/shelf_static.dart';
 import 'package:shelf_gzip/shelf_gzip.dart';
+import 'package:shelf_static/shelf_static.dart';
 
 void main() {
   // Assumes the server lives in bin/ and that `pub build` ran
   var pathToBuild =
-      join(dirname(Platform.script.toFilePath()), '..', 'build/web');
+  join(dirname(Platform.script.toFilePath()), '..', 'build/web');
   var staticHandler =
-      createStaticHandler(pathToBuild, defaultDocument: 'index.html');
+  createStaticHandler(pathToBuild, defaultDocument: 'index.html');
 
   var portEnv = Platform.environment['PORT'];
   var port = portEnv == null ? 9999 : int.parse(portEnv);
 
-  runZoned(() async {
+  runZonedGuarded(() async {
     var handler = const shelf.Pipeline()
-        .addMiddleware(gzipMiddleware)
-        .addHandler(staticHandler);
+            .addMiddleware(gzipMiddleware)
+            .addHandler(staticHandler);
 
     await shelf_io.serve(handler, '0.0.0.0', port);
 
     print("Serving $pathToBuild on port $port");
-  }, onError: (e, stackTrace) => print('Server error: $e $stackTrace'));
+  }, (e, stackTrace) => print('Server error: $e $stackTrace'));
 }
 ```
 
+## When not compress 
+
+The `gzip` encoding won't be applied if:
+
+- The `Content-Type` is for an already
+compressed type. See `isAlreadyCompressedContentType`.
+
+
+- A small response body (length < 512) will not benefit from being compressed.
```

Comment:
Why not allow this to be overriden?

Reply:
There's no statistical benefit to compress a body smaller than 512 for Gzip.

Reply:
Yes, but users might want to make this larger.

Reply:
I have added the parameter `minimalGzipContentLength`, to allow that.